kong/tools/queue.lua

```diff
@@ -539,8 +539,9 @@ local function enqueue(self, entry)
   -- The queue should not be full if we are running into this situation.
   -- Since the dropping logic is complcated,
   -- further maintenancers might introduce bugs,
-  -- so I added this assertion to detect this kind of bug early
-  -- (it's better to crash early than leak memory).
+  -- so I added this assertion to detect this kind of bug early.
+  -- It's better to crash early than leak memory
+  -- as analyze memory leak is hard.
   assert(
     -- bracing the function call to get the first return value only
     internal_is_full(self) == false,
```

Comment:
Would it not be better to test `internal_can_enqueue` here?

The comment could be more concise, if you feel that a comment is warranted at all.
```suggestion
-- assert that enough space is available on the queue now
```

Reply:
fixed

Reply:
https://github.com/Kong/kong/pull/13164/commits/9ad0141ce3c5cecfe4fe5cfb777b27ea20840693